sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -67,6 +80,15 @@ function getContentType(body: any): string {
     return "application/octet-stream";
   }
 
+  if (typeof body === "string") {
+    try {
+      JSON.parse(body);
+      return "application/json; charset=UTF-8";
+    } catch (error: any) {
+      // If we fail to parse the body, it is not json
+      return "text/plain; charset=UTF-8";
```

Comment:
I want to be cautious here and highlight that not all non-json strings are text/plain so I am not sure if we want to have that default here. Same argument can be used for the JSON case. Generally speaking, there are cases where the content-type may not be known and the server will need to figure it out by inspecting the payload. See https://www.rfc-editor.org/rfc/rfc9110.html#name-content-type

> A sender that generates a message containing content SHOULD generate a Content-Type header field in that message unless the intended media type of the enclosed representation is unknown to the sender. If a Content-Type header field is not present, the recipient MAY either assume a media type of "application/octet-stream" ([[RFC2046](https://www.rfc-editor.org/rfc/rfc9110.html#RFC2046)], [Section 4.5.1](https://www.rfc-editor.org/rfc/rfc2046#section-4.5.1)) or examine the data to determine its type.

Reply:
I am hoping we don't end up in situations with not ideal defaults such as https://github.com/Azure/azure-sdk-for-js/pull/28054#discussion_r1427312986

Reply:
Thanks for referencing it! So according to the RFC we could just leave the content-type undefined if it was not provided. @qiaozha what do you think about this? maybe the fallback case would just be to leave content-type as undefined and send the body as we have it?

Reply:
There are many JSON-like payloads that are not application/json such as JSON schemas and JSON-LD so I am against the default content type here.

If we need to do this to preserve backward-compatability, let's open an issue to track correcting this in a future major version bump.

Reply:
Lets file an issue to track this, I think we'll have to keep it this way to avoid a runtime breaking change. @qiaozha would you mind filing an issue for this?

Reply:
issue created here https://github.com/Azure/azure-sdk-for-js/issues/28487